process_report/processors/lenovo_processor.py

```diff
@@ -0,0 +1,30 @@
+from dataclasses import dataclass
+
+
+from process_report.invoices import invoice
+from process_report.processors import processor
+
+
+@dataclass
+class LenovoProcessor(processor.Processor):
+    LENOVO_SU_TYPES = ["OpenShift GPUA100SXM4", "OpenStack GPUA100SXM4"]
+    SU_CHARGE_MULTIPLIER = 1
+
+    def _prepare(self):
+        self.data = self.data[
```

Comment:
Yes. After this statement, `self.data` will contain a subset of itself. This doesn't change the dataframe that `self.data` originally referred to, which means if other invoices was given the original dataframe prior to this statement, it would not affect them in anyway.

I can see how you might not want the code to be written like this regardless. I could rewrite the code so that invoices never reassign to `self.data`, though that will require some significant changes.

Reply:
@QuanMPhm Yes, based on the discussion about all Processors eventually being applied to the same Dataframe, we don't want Processors to be doing that sort of filtering as that would mean other Processors can't be applied afterwards.

If it makes things easier for you, I would be okay with Invoices in `_prepare_export` assigning a `self.export_data` that contains the Dataframe to be exported.

Reply:
@knikolla I just realized you were referring to this change as if it was in the Lenovo Processor. Based on the current version of the PR, this change is in the Lenovo **Invoice**. If its fine for the Invoice Classes to do this "no-data-modified" filtering, then I will keep the PR as it is.

Reply:
Ok for now, but in the future I think it makes much more sense for Invoices not to reassign `self.data`, but to instead assign a new variable `self.export_data`. This can be implemented in the base Invoice class.

Reply:
I'll make a PR at the end of the refactoring have all invoices perform filtering through `self.export_data`